rewrite-pattern.js

```diff
@@ -387,32 +414,36 @@ const computeCharacterClass = (characterClassItem, regenerateOptions) => {
 	}
 
 	if (characterClassItem.negative && data.maybeIncludesStrings) {
-		throw new SyntaxError("Cannot negate set containing strings");
+		throw new SyntaxError('Cannot negate set containing strings');
 	}
 
 	return data;
 }
 
-const processCharacterClass = (characterClassItem, regenerateOptions) => {
+const processCharacterClass = (
+	characterClassItem,
+	regenerateOptions,
+	computed = computeCharacterClass(characterClassItem, regenerateOptions),
+) => {
 	const negative = characterClassItem.negative;
-	const { singleChars, transformed, longStrings } = computeCharacterClass(characterClassItem, regenerateOptions);
+	const { singleChars, transformed, longStrings } = computed;
 	if (transformed) {
 		const setStr = singleChars.toString(regenerateOptions);
 
 		if (negative) {
 			if (config.useUnicodeFlag) {
-				update(characterClassItem, `[^${setStr[0] === "[" ? setStr.slice(1, -1) : setStr}]`)
+				update(characterClassItem, `[^${setStr[0] === '[' ? setStr.slice(1, -1) : setStr}]`)
 			} else {
 				update(characterClassItem, `(?!${setStr})[\\s\\S]`)
 			}
 		} else {
-			const hasEmptyString = longStrings.has("");
+			const hasEmptyString = longStrings.has('');
 			const pieces = Array.from(longStrings).sort((a, b) => b.length - a.length);
```

Comment:
Drive-by: like in `u` mode, we think in terms of code points, not "JS characters" in `v` mode. So we technically need to sort by code point length: `Array.from(b).length - Array.from(a).length` (or `[...b].length - [...a].length`)

Reply:
Uhm, is this observable in any way? I think we need sorting only when `A` is a prefix of `B`, but in that case `Array.from(A).length < Array.from(B).length` will be the same as `A.length < B.length`.

(Here we are sorting a potentially very big list, so it's better to keep the comparator fast&simple :sweat_smile:)